### ISSUE TITLE
Renaming a print format breaks the "view this in your browser" link for emails already sent

### ISSUE THREAD

**Issue by pcharlc** [frappe/erpnext#4925]
We've discovered the following scenario:

- a document is generated (eg sales invoice / quotation)
- it is emailed to the customer  using a particular print format (in our scenario, we've defined a particular print format to include additional information, such as "this is for services during the month of ...")
- the email includes "view this in your browser" link, which the customer can use to view a web version of the doc.

- if the print format is then later renamed, and the customer subsequently goes back to that email and uses the "view this in your browser" link, it will no longer work -- instead customer will see a traceback ultimately showing the print format cannot be found.

This probably happens because the "view" link contains format= parameter that refers to the print format by name, when it should probably be using a fixed id.

**Comment by rmehta** [comment]
This is very rare. In any case we don't save the old name.

**Comment by pcharlc** [comment]
Respectfully, I disagree.  We ran into this within 6 weeks of starting to use ERPnext, in a print format that included a generated credit card payment link.  After we renamed the print format (purely to make our naming be more consistent over various formats), we discovered we could no longer get paid on invoices that had already been e-mailed out when clients tried to access the invoice by clicking on the link embedded in the emails they had received.

Objects (like print formats) that are referenced by URLs that will be sticking around (eg in sent emails) should be getting referenced by something that is immutable -- like id/uuid -- not by something that can be  changed at any moment (like name)
